src/UniversalRewardsDistributor.sol

```diff
@@ -122,12 +122,16 @@ contract UniversalRewardsDistributor is IUniversalRewardsDistributor {
             ErrorsLib.INVALID_PROOF_OR_EXPIRED
         );
 
+        require(claimable >= claimed[account][reward], ErrorsLib.ROOT_MISSCONFIGURED);
+
         amount = claimable - claimed[account][reward];
 
         require(amount > 0, ErrorsLib.ALREADY_CLAIMED);
 
         claimed[account][reward] = claimable;
 
+        require(ERC20(reward).balanceOf(address(this)) >= amount, ErrorsLib.NOT_ENOUGH_FUNDS_IN_URD);
+
```

Comment:
I'm against adding this check, I feel like this should be handled at the ERC20 level

Reply:
This is a way to have a unified error message on all URD, considering that the ERC20 can revert for other reasons (permissions etc). 
I agree with you that it's redundant, but it is much cleaner IMO.

Reply:
Not sure about much cleaner. Is it significantly easier to use though ? In which case I could see why we would want to keep it, even if it is redundant

Reply:
I have no good reason to keep it to be honest.

Reply:
Ive removed the line, but I let the discussion open if someone has a good argument for the balance early check.

Reply:
Good argument: we directly knows that there's not enough funds which removes debugging time

no?

Reply:
up @julien-devatom

Reply:
I also feel like it's redundant but I'm not blocking

Reply:
let's forget then